dht/token.cc

```diff
@@ -271,4 +271,21 @@ dht::token find_first_token_for_shard(
     }
 }
 
+size_t
+compaction_group_of(unsigned most_significant_bits, const token& t) {
+    if (!most_significant_bits) {
+        return 0;
+    }
+    switch (t._kind) {
+        case token::kind::before_all_keys:
+            return 0;
+        case token::kind::after_all_keys:
+            return (1 << most_significant_bits) - 1;
+        case token::kind::key:
+            uint64_t adjusted = unbias(t);
```

Comment:
this should not compile. see https://godbolt.org/z/s6foE9jM1

Reply:
maybe inlining gets around it.

this was copied from https://github.com/scylladb/scylladb/blob/ebc100f74fc2d8946cf4810499ce1f0c70ba5ea8/dht/token.cc#L195-L197
that compiles happily since 2015 (7b1e78ffcd711ce03ee56798eabe468330911c9f)

Reply:
it's not related to inline. it's just we don't have a branch **after** `case token::kind::key:`, so the statement for initialization of `uint64_t adjusted` won't be "jumped over" if the switch-case clause jumps to that branch. and hence we don't have the error like
```
<source>:12:9: error: jump to case label
   12 |         default:
      |         ^~~~~~~
<source>:9:13: note:   crosses initialization of 'int j'
    9 |         uint64_t adjusted = unbias(t);
      |                  ^
```

Reply:
ah, makes sense.
By convention, we don't define complimentary default cases when switching over enum values to help the compiler warn us about unhandled values.
We also have the `__builtin_unreached()` call outside the switch.

Reply:
thank you, Benny. that's a good practice. and i will follow it in future.